prov/efa/src/efa_domain.c

```diff
@@ -140,7 +140,9 @@ static int efa_domain_init_rdm(struct efa_domain *efa_domain, struct fi_info *in
 {
 	int err;
 
-	efa_shm_info_create(info, &efa_domain->shm_info);
+	efa_domain->shm_info = NULL;
+	if (efa_env.enable_shm_transfer)
+		efa_shm_info_create(info, &efa_domain->shm_info);
```

Comment:
Consider some logging, e.g.
```suggestion
	if (efa_env.enable_shm_transfer)
		efa_shm_info_create(info, &efa_domain->shm_info);
	else
		EFA_INFO(FI_LOG_CORE, "Skipping fi_info allocation for SHM.\n");
```

Reply:
done.